cpp/src/arrow/compute/api_scalar.h

```diff
@@ -420,6 +420,30 @@ struct ARROW_EXPORT Utf8NormalizeOptions : public FunctionOptions {
   Form form;
 };
 
+class ARROW_EXPORT RandomOptions : public FunctionOptions {
+ public:
+  enum Initializer { SystemRandom, Seed };
+
+  static RandomOptions FromSystemRandom(int64_t length) {
+    return RandomOptions{length, SystemRandom, 0};
+  }
+  static RandomOptions FromSeed(int64_t length, uint32_t seed) {
+    return RandomOptions{length, Seed, seed};
+  }
+
+  RandomOptions(int64_t length, Initializer initializer, uint32_t seed);
+  RandomOptions();
+  constexpr static char const kTypeName[] = "RandomOptions";
+  static RandomOptions Defaults() { return RandomOptions(); }
+
+  /// The length of the array returned. Negative is invalid.
+  int64_t length;
+  /// The type of initialization for random number generation - system or provided seed.
+  Initializer initializer;
+  /// The seed value used to initialize the random number generation.
+  uint32_t seed;
```

Comment:
Do we want to make this `uint64_t` to allow passing more entropy? Or would the higher bits be ignored?

Reply:
Done. I don't know why it was changed to `uint32_t`, maybe @cyb70289 had a good reason. I made it `int64_t` initially, should have been `uint64_t`.

Reply:
It was `uint32_t` to be consistent with system random seed, also references numpy.random.seed.
But as we have changed to use pcg64 for system random seed, it looks better be also use uint64 here.